File: core/detect.go
package core

import (
	"bufio"
	"os"
	"strings"
)

type Distro string

// supported distros
const (
	Arch    Distro = "arch"
	Debian  Distro = "debian"
	Fedora  Distro = "fedora"
	Unknown Distro = "unknown"
)

type SystemInfo struct {
	Distro Distro
	PkgMgr string
	Name   string
}

func Detect() SystemInfo {
	f, err := os.Open("/etc/os-release")
	if err != nil {
		return SystemInfo{Distro: Unknown}
	}
	defer f.Close()

	vals := map[string]string{}
	scanner := bufio.NewScanner(f)
	for scanner.Scan() {
		line := scanner.Text()
		parts := strings.SplitN(line, "=", 2)
		if len(parts) == 2 {
			vals[parts[0]] = strings.Trim(parts[1], `"`)
		}
	}

	id := strings.ToLower(vals["ID"])
	idLike := strings.ToLower(vals["ID_LIKE"])
	name := vals["PRETTY_NAME"]

	switch {
	case id == "arch" || strings.Contains(idLike, "arch"):
		return SystemInfo{Distro: Arch, PkgMgr: "pacman", Name: name}
	case id == "fedora" || strings.Contains(idLike, "fedora"):
		return SystemInfo{Distro: Fedora, PkgMgr: "dnf", Name: name}
	case id == "debian" || id == "ubuntu" || id == "linuxmint" ||
		strings.Contains(idLike, "debian") || strings.Contains(idLike, "ubuntu"):
		return SystemInfo{Distro: Debian, PkgMgr: "apt", Name: name}
	default:
		return SystemInfo{Distro: Unknown, PkgMgr: "", Name: name}
	}
}

File: core/detect.py
import os
import shutil 

def detect_distro():
    values = {}
    
    with open("/etc/os-release") as f:
        for line in f:
            if "=" in line:
                key, val = line.strip().split("=", 1)
                values[key] = val.strip('"').lower()
    
    distro_id = values.get("ID", "")
    id_like = values.get("ID_LIKE", "")
    
    if distro_id == "arch" or "arch" in id_like:
        return "arch"
    
    if distro_id in ("debian", "ubuntu", "linuxmint", "pop") or \
        "debian" in id_like or "ubuntu" in id_like:
        return "debian"
    
    if distro_id == "fedora" or "fedora" in id_like:
        return "fedora"
    
    return "unknown"

def detect_shell():
    return os.environ.get("SHELL", "").split("/")[-1]

def detect_terminals():
    home = os.path.expanduser("~")
    terminals = {
        "alacritty": f"{home}/.config/alacritty",
        "kitty":     f"{home}/.config/kitty",
        "ghostty":   f"{home}/.config/ghostty",
        "wezterm":   f"{home}/.config/wezterm",
    }
    found = []
    for name, path in terminals.items():
        if os.path.exists(path):
            found.append(name)
    return found


def detect_fetch():
    fetches = ["fastfetch", "neofetch", "pfetch", "screenfetch"]
    found = []
    for fetch in fetches:
        if shutil.which(fetch):
            found.append(fetch)
    return found 